### DISCUSSION TITLE
consuming hangs at empty offset caused by aborted transaction

### DISCUSSION THREAD

**Discussion by rickif** [edenhill/librdkafka#3766] [Q&A]
When there are aborted Kafka transactions in the topic, the messages may be with discontinuous offsets. 
Suppose we have kafka message as following:
```
offset 10: with message
offset 11: with message
offset 12: empty message caused by aborted transaction 
offset 13: empty message caused by aborted transaction 
```
The `query_watermark_offsets` would return low/high as 10/13.
But, the `Message *consume (int timeout_ms)` would hangs at offset 12 and returns `RdKafka::ERR__TIMED_OUT`.
What I expect is that the `consume` could reach the offset 13 and return empty message.
Is it by design or something wrong with my config?

**Comment by edenhill** [comment]
The abort marker messages (12 and 13) will be ignored and not passed to the application, so unless there are more messages being produced to the partition the consumer will not return any new messages (and thus time out according to your timeout argument)
